python_modules/dagster/dagster/_core/utils.py

```diff
@@ -127,6 +128,35 @@ def parse_env_var(env_var_str: str) -> Tuple[str, str]:
 class InheritContextThreadPoolExecutor(ThreadPoolExecutor):
     """A ThreadPoolExecutor that copies over contextvars at submit time."""
 
+    def __init__(
+        self,
+        max_workers: int | None = None,
+        thread_name_prefix: str = "",
+        initializer: Callable[..., object] | None = None,
+        initargs: tuple[Any, ...] = ...,
+    ) -> None:
+        super().__init__(max_workers, thread_name_prefix, initializer, initargs)
+        self._futures = []
+
     def submit(self, fn, *args, **kwargs):
         ctx = copy_context()
-        return super().submit(ctx.run, fn, *args, **kwargs)
+        new_future = super().submit(ctx.run, fn, *args, **kwargs)
+        self._futures = [
+            future for future in self._futures if not future.done()
+        ]  # clean up done futures
+        self._futures.append(new_future)
+        return new_future
+
+    @property
+    def max_workers(self) -> int:
+        return self._max_workers
+
+    @property
+    def num_running_futures(self) -> int:
+        return (
+            len([future for future in self._futures if not future.done()]) - self.num_queued_futures
```

Comment:
Maybe extract out calculating the length of `self._futures` into a separate method?

Reply:
What's the reasoning behind doing that? Doesn't seem like we'll be reusing it

Reply:
Just to document the code better. Maybe just need to extract it as a variable in the function. Up to you, this feedback is non-blocking

```suggestion
    @property
    def num_running_futures(self) -> int:
        num_unfinished_futures = len([future for future in self._futures if not future.done()])

        return num_unfinished_futures - self.num_queued_futures
```

Reply:
I see. Makes sense.